assets/js/base/components/text-input/style.scss

```diff
@@ -6,14 +6,15 @@
 .wc-block-text-input label {
 	position: absolute;
 	transform: translateX(#{$gap}) translateY(#{$gap-small});
+	transform-origin: top left;
```

Comment:
I was going to rewrite this later by introducing a `useLayoutEffect` that calculates the width dynamically because, In the initial code I wrote, this did not work for me, I had translate happen before scale, I see you managed to get it to work, not sure what changed browser wise, but this is much better.